src/main/drivers/bus_i2c_hal.c

```diff
@@ -102,7 +102,7 @@ uint16_t i2cGetErrorCounter(void)
 // Blocking write
 bool i2cWrite(I2CDevice device, uint8_t addr_, uint8_t reg_, uint8_t data)
 {
-    if (device == I2CINVALID || device > I2CDEV_COUNT) {
+    if (device == I2CINVALID || device >= I2CDEV_COUNT) {
```

Comment:
Won't this break the case when we have the maximum I2C devices as device allow?

Reply:
Should not break anything. Check next few lines. There is `I2C_HandleTypeDef *pHandle = &i2cDevice[device].handle;` and here is the definiton `i2cDevice_t i2cDevice[I2CDEV_COUNT];`. So `device` should never be `I2CDEV_COUNT`.